spec/controllers/v0/account_controls_controller_spec.rb

```diff
@@ -0,0 +1,288 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe V0::AccountControlsController, type: :controller do
+  let(:service_account_access_token) { build(:service_account_access_token) }
+  let(:user_account) { create(:user_account) }
+  let(:locked) { false }
+  let(:logingov_user_verification) { create(:logingov_user_verification, user_account:, locked:) }
+  let(:idme_user_verification) { create(:idme_user_verification, user_account:, locked:) }
+  let(:dslogon_user_verification) { create(:dslogon_user_verification, user_account:, locked:) }
+  let(:mhv_user_verification) { create(:mhv_user_verification, user_account:, locked:) }
+
+  before do
+    allow_any_instance_of(V0::AccountControlsController).to receive(:authenticate_service_account).and_return(true)
+    controller.instance_variable_set(:@service_account_access_token, service_account_access_token)
+  end
+
+  describe 'GET csp_index' do
+    let(:icn) { user_account.icn }
+    let(:icn_param) { icn }
+
+    context 'without an ICN param' do
+      let(:icn_param) { nil }
+      let(:expected_error) { 'The required parameter "icn", is missing' }
+
+      it 'returns a bad request error' do
+        get :csp_index, params: { icn: icn_param }
+
+        expect(response).to have_http_status(:bad_request)
+        error = JSON.parse(response.body)['errors'].first
+        expect(error['detail']).to eq(expected_error)
+      end
+    end
+
+    context 'with an ICN param' do
+      context 'when a record is not found' do
+        let(:icn_param) { 'some-bad-icn' }
+        let(:expected_error) { "User CSP Index not found. ICN:#{icn_param}" }
+
+        it 'returns a not found error' do
+          get :csp_index, params: { icn: icn_param }
+
+          expect(response).to have_http_status(:not_found)
+          error = JSON.parse(response.body)['error']
+          expect(error).to eq(expected_error)
+        end
+      end
+
+      context 'when one or more User Verifications are found' do
+        let(:expected_csp_uuids) do
+          [logingov_user_verification.logingov_uuid,
+           idme_user_verification.idme_uuid,
+           dslogon_user_verification.dslogon_uuid,
+           mhv_user_verification.mhv_uuid]
+        end
+
+        before do
+          logingov_user_verification
```

Comment:
Don't do this here, instead, when you define the `let`, make it `let!` and it will load the object